converter/src/main/java/gov/cms/qpp/conversion/model/error/Detail.java

```diff
@@ -31,42 +64,26 @@ public Detail() {
 	 * Copy constructor
 	 */
 	public Detail(Detail detail) {
-		this(detail.getMessage(), detail.getPath(), detail.getValue(), detail.getType());
-	}
-
-	/**
-	 * Constructs a {@code Detail} with just a description.
-	 *
-	 * @param text A description of the error.
-	 */
-	public Detail(String text) {
-		this.message = text;
+		setErrorCode(detail.getErrorCode());
+		setMessage(detail.getMessage());
+		setPath(detail.getPath());
+		setValue(detail.getValue());
+		setType(detail.getType());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setType" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1699)